File: Dataset/OOP_prepared/Python/GeometricObject_trace_1.py
import math
from abc import ABC, abstractmethod

class GeometricObject(ABC):
    @abstractmethod
    def get_area(self):
        pass

    @abstractmethod
    def get_perimeter(self):
        pass

class Circle(GeometricObject):
    def __init__(self, radius=0.0):
        self.radius = radius

    def get_radius(self):
        return self.radius

    def set_radius(self, radius):
        self.radius = radius

    def __str__(self):
        return f"Circle[radius={self.radius}]"

    def get_area(self):
        return self.radius * self.radius * math.pi

    def get_perimeter(self):
        return self.radius * 2 * math.pi

class Rectangle(GeometricObject):
    def __init__(self, width=0.0, length=0.0):
        self.width = width
        self.length = length

    def get_width(self):
        return self.width

    def set_width(self, width):
        self.width = width

    def get_length(self):
        return self.length

    def set_length(self, length):
        self.length = length

    def __str__(self):
        return f"Rectangle[width={self.width}, length={self.length}]"

    def get_area(self):
        return self.width * self.length

    def get_perimeter(self):
        return (self.width + self.length) * 2

if __name__ == "__main__":
    c = Circle(3.5)
    print("Area:", c.get_area())

File: Dataset/Advanced/Test/OOP_prepared/Java/GeometricObject_trace_1.java
package GeometricObject;

public interface GeometricObject {
    public double getArea();
    public double getPerimeter();
}

public class Circle implements GeometricObject{

    private double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    public Circle() {
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }

    @Override
    public String toString() {
        return "Circle[" + "radius=" + radius + ']';
    }
    
    @Override
    public double getArea() {
        return radius*radius*Math.PI;
    }

    @Override
    public double getPerimeter() {
        return radius*2*Math.PI;
    }
    
}

public class Rectangle implements GeometricObject{
    private double width;
    private double length;

    public Rectangle() {
    }

    public Rectangle(double width, double length) {
        this.width = width;
        this.length = length;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) {
        this.length = length;
    }

    @Override
    public String toString() {
        return "Rectangle[" + "width=" + width + ",length=" + length + ']';
    }

    @Override
    public double getArea() {
        return length*width;
    }

    @Override
    public double getPerimeter() {
        return (width+length)*2;
    }
}

public class TestMain {
    public static void main(String[] args) {
        GeometricObject c1 = new Circle(3.5);
        System.out.println("Area: " + c1.getArea());
    }
}